### PR TITLE
Allow http_method to be specified for searches

### PR DESCRIPTION
**Pull request by mgodwin**

### CODE REVIEW

**Comment by mgodwin** [comment]
Hey @DubHunt will you give me a quick set of 👀 on this?

File: lib/rets/base/core.rb [hunk 1]
```diff
@@ -235,6 +235,7 @@ def get_object(args, &block)
       # @option args [Integer, Optional] :read_timeout How long to wait for data from the socket before giving up
       # @option args [Integer, Optional] :open_timeout How long to wait for connection to open before giving up
       # @option args [Boolean, Optional] :disable_stream Disables the streaming setup for data and instead loads it all and then parses
+      # @option args [Boolean, Optional] :http_method Specifies what http method to use for the request - GET/POST - GET by default
```
Comment (DubHunt) [inline lib/rets/base/core.rb]: `s/Boolean/String/`?

File: lib/rets/base/core.rb [hunk 2]
```diff
@@ -254,7 +255,8 @@ def search(args, &block)
           raise RETS::CapabilityNotFound.new("Cannot find URL for Search call")
         end
 
-        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout]}
+        http_method = args[:http_method] || 'GET'
+        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout], method: args[:http_method]}
```
Comment (DubHunt) [inline lib/rets/base/core.rb]: You're setting the `method` option here to `args[:http_method]` instead of `http_method`.

File: lib/rets/http.rb [hunk 3]
```diff
@@ -207,8 +200,24 @@ def request(args, &block)
         http.ca_path = @config[:http][:ca_path] if @config[:http][:ca_path]
       end
 
+      body_data = nil
+      request = nil
+      if args[:http_method].to_s.downcase == 'post'
+        request = Net::HTTP::Post.new request_uri, headers
+        body_data = args[:params] if args[:params]
```
Comment (DubHunt) [inline lib/rets/http.rb]: nit: why check `if args[:params]`?

File: lib/rets/base/core.rb (hunk 1 above)
Comment (mgodwin) [reply lib/rets/base/core.rb]: Wow thx, so sloppy of me 😲

File: lib/rets/base/core.rb (hunk 2 above)
Comment (mgodwin) [reply lib/rets/base/core.rb]: I wish github had a :yakety: emoji

File: lib/rets/http.rb (hunk 3 above)
Comment (mgodwin) [reply lib/rets/http.rb]: 🤔 I'm not sure, good point!

**Comment by mgodwin** [comment]
@DubHunt fixed comments - one more time? Then we'll get this merged 👍

File: lib/rets/base/core.rb [hunk 4]
```diff
@@ -235,6 +235,7 @@ def get_object(args, &block)
       # @option args [Integer, Optional] :read_timeout How long to wait for data from the socket before giving up
       # @option args [Integer, Optional] :open_timeout How long to wait for connection to open before giving up
       # @option args [Boolean, Optional] :disable_stream Disables the streaming setup for data and instead loads it all and then parses
+      # @option args [Symbol, Optional] :http_method Specifies what http method to use for the request - GET/POST - GET by default
```
Comment (DubHunt) [inline lib/rets/base/core.rb:238]: Is it a symbol? You default to `'GET'` below.

File: lib/rets/base/core.rb [hunk 5]
```diff
@@ -254,7 +255,8 @@ def search(args, &block)
           raise RETS::CapabilityNotFound.new("Cannot find URL for Search call")
         end
 
-        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout]}
+        http_method = args[:http_method] || 'GET'
+        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout], http_method: args[:http_method]}
```
Comment (DubHunt) [inline lib/rets/base/core.rb:258]: Still not using `http_method`? 😄

**Comment by mgodwin** [comment]
Cool, merging & closing this down.

**Pull request merged**